code/game/objects/items/robot/robot_items/robot_furnace.dm

```diff
@@ -0,0 +1,100 @@
+//Totally not copypasted from its machinery version stop asking so many questions
+#define PORTOSMELTER_MINIMUM_POWER_REQUIRED 100
+#define PORTOSMELTER_POWER_DRAIN rand(50,100)
+
+/obj/item/weapon/storage/bag/ore/furnace
+	name = "portable ore processor"
+	icon_state = "portsmelter"
+	actions_types = list(/datum/action/item_action/toggle_furnace)
+	var/event_key = null
+	var/datum/materials/ore
+	var/list/recipes[0]
+	var/on = FALSE
+	var/sheets_per_tick = 100
+
+/datum/action/item_action/toggle_furnace
+	name = "Toggle Ore Furnace"
+
+/datum/action/item_action/toggle_furnace/Trigger()
+	var/obj/item/weapon/storage/bag/ore/furnace/F = target
+	var/mob/user = usr
+
+	if(!usr)
+		if(!ismob(F.loc))
+			return
+		user = F.loc
+
+	if(!istype(F) || !user)
+		return
+
+	F.smelt_contents(user)
+
+/obj/item/weapon/storage/bag/ore/furnace/New()
+	. = ..()
+	ore = new
+	for(var/recipe in typesof(/datum/smelting_recipe) - /datum/smelting_recipe)
+		recipes += new recipe()
```

Comment:
Parenthesis aren't necessary and I personally find them a bit confusing but it's not important at all.